src/compiler/scala/tools/nsc/Driver.scala

```diff
@@ -65,5 +66,5 @@ abstract class Driver {
     !reporter.hasErrors
   }
 
-  def main(args: Array[String]): Unit = System.exit(if (process(args)) 0 else 1)
+  def main(args: Array[String]): Unit = if (!process(args)) throw SystemExit(1)
```

Comment:
This means when I run `scalac` I get a stack trace, which cannot be the intention. If you turn off the stack trace for the exception, you always get a line with the exception type and whatever message. That can be minimized but not eliminated? 

You could get clever with the class name, call it `exited` and leave it in the empty package with a null message.

The entry point for tools is `process`, not `main`.

Reply:
I haven't looked at this particular code, but just in general, calling `System.exit` from a `main` method seems normal to me— I think it's the ones that aren't in `main` methods that we were trying to get rid of

Reply:
ok 👍